wagtail_localize/segments/ingest.py

```diff
@@ -226,10 +227,23 @@ def ingest_segments(original_obj, translated_obj, src_locale, tgt_locale, segmen
         field_name, segment = segment.unwrap()
         segments_by_field_name[field_name].append(segment)
 
-    for field_name, field_segments in segments_by_field_name.items():
+    for field in get_translatable_fields(translated_obj.__class__):
+
+        field_name = field.field_name
+        field_segments = segments_by_field_name[field_name]
         field = translated_obj.__class__._meta.get_field(field_name)
 
-        if hasattr(field, "restore_translated_segments"):
+        if not field_segments:
+
+            if isinstance(field, (models.TextField, models.CharField)):
```

Comment:
Maybe we should synchronise all translatable fields before restoring translations? We already do this for streamfield